File: ios/RBDarts/Features/StandaloneGame/StandaloneGameDiagnostics.swift
import Foundation

public struct StandaloneGameDiagnostics {
    private let reporter: DiagnosticsReporting

    public init(reporter: DiagnosticsReporting = PrivacySafeDiagnostics()) {
        self.reporter = reporter
    }

    public func recordInvalidScore(inning: Int) {
        reporter.record(DiagnosticEvent(
            name: "standalone_invalid_score",
            severity: .warning,
            attributes: ["inning": "\(inning)"]
        ))
    }

    public func recordScoreAccepted(inning: Int) {
        reporter.record(DiagnosticEvent(
            name: "standalone_score_accepted",
            severity: .debug,
            attributes: ["inning": "\(inning)"]
        ))
    }

    public func recordRecovery() {
        reporter.record(DiagnosticEvent(name: "standalone_recovery", severity: .info, attributes: [:]))
    }

    public func recordCompletedGame() {
        reporter.record(DiagnosticEvent(name: "standalone_completed", severity: .info, attributes: [:]))
    }
}

File: android/app/src/main/java/com/rbdarts/feature/standalonegame/StandaloneGameDiagnostics.kt
package com.rbdarts.feature.standalonegame

import com.rbdarts.core.observability.DiagnosticEvent
import com.rbdarts.core.observability.DiagnosticSeverity
import com.rbdarts.core.observability.DiagnosticsReporter
import com.rbdarts.core.observability.PrivacySafeDiagnostics

class StandaloneGameDiagnostics(
    private val reporter: DiagnosticsReporter = PrivacySafeDiagnostics()
) {
    fun recordInvalidScore(inning: Int) {
        reporter.record(
            DiagnosticEvent(
                name = "standalone_invalid_score",
                severity = DiagnosticSeverity.Warning,
                attributes = mapOf("inning" to "$inning")
            )
        )
    }

    fun recordScoreAccepted(inning: Int) {
        reporter.record(
            DiagnosticEvent(
                name = "standalone_score_accepted",
                severity = DiagnosticSeverity.Debug,
                attributes = mapOf("inning" to "$inning")
            )
        )
    }

    fun recordRecovery() {
        reporter.record(DiagnosticEvent(name = "standalone_recovery", severity = DiagnosticSeverity.Info))
    }

    fun recordCompletedGame() {
        reporter.record(DiagnosticEvent(name = "standalone_completed", severity = DiagnosticSeverity.Info))
    }
}
